flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -165,20 +167,10 @@ class ReachCollector {
       popFn(rop);
       return;
     }
-    if (auto rop = mlir::dyn_cast<IterWhileOp>(op)) {
-      popFn(rop);
-      return;
-    }
     if (auto rop = mlir::dyn_cast<fir::IfOp>(op)) {
       popFn(rop);
       return;
     }
-    if (auto box = mlir::dyn_cast<EmboxOp>(op)) {
```

Comment:
Why is EmboxOp being dropped?

Reply:
Couldn't find a place where it was used. We can keep it in fir-dev. Anyway we have to do a second upstreaming step.

Reply:
@jeanPerier Should we keep `EmboxOp` here.

Reply:
Looks like I added it to fix some NAG bugs.